deepspeed/runtime/zero/stage3.py

```diff
@@ -1855,7 +1855,7 @@ def _post_step(self, timer_names):
         if self.swap_optimizer:
             self.optimizer_swapper.log_timers()
 
-        self.invalidate_secondary_tensor()
+        # self.invalidate_secondary_tensor() # given that we want hpz in forward pass when no_grad is set, we need to keep the secondary tensor
```

Comment:
I am not sure about this. Could the weight update by step not invalidate the cached secondary tensor values?